### ISSUE TITLE
SomaC center

### ISSUE THREAD

**Issue by mgeplf** [BlueBrain/NeuroM#231]
Hello,
The documentation states:
```
    Type C: multiple points soma
    Represented by a contour.
    Reference: neuromorpho.org
    The first point constitutes the center of the soma,
    '''
```

But I'm having problems confirming that the first point is the center of the soma on neuromorpho.org, could you please provide a pointer to this?

**Comment by juanchopanza** [comment]
I can't find it either. I found this:

> Soma can be a single point or more than one point. When the soma is encoded as one line in the SWC, it is interpreted as a "sphere". When it is encoded by more than 1 line, it could be a set of tapering cylinders (as in some pyramidal cells) or even a 2D projected contour ("circumference"). 

@lidakanari could you comment on this?

**Comment by lidakanari** [comment]
I used this definition: http://neuromorpho.org/neuroMorpho/SomaFormat.html which apparently has been removed!!!

**Comment by juanchopanza** [comment]
We can remove the reference to neuromorpho and say this is how we interpret a multiple point soma, in the absence of any means to know what it is supposed to represent. We could also think of some heuristics to guess what three or more points represent.

**Comment by lidakanari** [comment]
So we should define the three types of soma by ourselves? (That seems like the only solution in the absence of a meaningful definition in the web).

**Comment by lidakanari** [comment]
Update: new location found! 

http://neuromorpho.org/SomaFormat.html

**Comment by mgeplf** [comment]
Cool, thanks for finding that.  I read the quoted string as saying that the soma center should be the average of the contour points, which is also how the h5 calculates it.  Right now, I don't think that is happening: it's just picking the first point and assuming it's the center which means the soma.center property is confusing.

**Comment by lidakanari** [comment]
It assumes that the first point contains those info. We definitely have to look into this and probably confirm it! I'd like to add this as a proper function but I'm still waiting for a concrete format for the soma structures.

**Comment by mgeplf** [comment]
I think I fixed it here: #233 

Also, I think it would be a good idea to name the soma classes more descriptively:
SomaA -> SomaSinglePoint
SomaB -> SomaThreePoint
SomaC -> SomaContour

**Comment by juanchopanza** [comment]
Fixed in #233.

**Comment by lidakanari** [comment]
I completely agree with @mgeplf !
